backend/main.py

```
@@ -830,6 +841,62 @@ async def delete_memory(project_id: str, memory_id: str):
+    try:
+        project = file_service.get_project_by_id(project_id)
+        if not project:
+            raise HTTPException(status_code=404, detail="Project not found")
+        content = (request.content or "").strip()
+        file_service.save_project_detail(project_id, content)
+        return {"status": "saved", "chars": len(content)}
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.error(f"Error saving project detail: {e}")
+        raise HTTPException(status_code=500, detail=f"Failed to save project detail: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)